src/components/AnnotationViewer.vue

```diff
@@ -2223,34 +2224,28 @@ export default class AnnotationViewer extends Vue {
           "pointer"
             ? "point"
             : "polygon";
-        this.annotationLayer.mode(selectionType);
+        this.interactionLayer.mode(selectionType);
         break;
       case "samAnnotation":
       case null:
       case undefined:
-        this.annotationLayer.mode(null);
+        this.interactionLayer.mode(null);
         break;
       default:
         logWarning(
           `${this.selectedToolConfiguration?.type} tools are not supported yet`,
         );
-        this.annotationLayer.mode(null);
+        this.interactionLayer.mode(null);
     }
+  }
 
-    if (this.selectedToolConfiguration?.type === "tagging") {
-      this.annotationLayer.geoOn(
-        geojs.event.mouseclick,
-        this.handleTaggingClick,
-      );
-    } else {
-      this.annotationLayer.geoOff(
-        geojs.event.mouseclick,
-        this.handleTaggingClick,
-      );
+  handleModeChange(evt: any) {
```

Comment:
Should the type be `IGeoJSMouseState` ?

Reply:
If I do that, I get a lot of typescript issues. I don't know why Thibault did it with `any`, but it seems to be a purposeful design choice on his part.

Reply:
Oh yeah, that's because GeoJS is no a Typescript library, so we don't have types for those events.  
We could add more types to `src/store/model.ts` but `any` is fine.